libraries/lib-wave-track/WaveClip.cpp

```diff
@@ -1281,6 +1280,8 @@ void WaveClip::ApplyStretchRatio(
    SetTrimLeft(0.);
    SetTrimRight(0.);
    SetSequenceStartTime(originalPlayStartTime);
+   mRawAudioTempo = *mProjectTempo;
+   mClipStretchRatio = targetRatio;
```

Comment:
This change is wrong for exception safety!

Please keep the finally happy-path updates as they were, inside the `Finally`.

`Flush` can throw, and we can't predict what overrides of `Invalidate` may throw.

Reply:
Also please preserve the assertion of the postcondition.